### PR TITLE
WIP: Test implementation

### PR DESCRIPTION
**Pull request by modev**
Since there weren't any test for this package, I just started to add some basic test cases with mocha and chai. I still don't know how to handle the auth tests properly, so I just built a test for the error scenario. Some ideas how to fix this? 

- [X] Auth
- [X] History
- [x] Invalidate
- [x] Profile
- [x] Refresh
- [x] Signout
- [X] Status
- [x] User
- [X] Username
- [x] Validate

### CODE REVIEW

File: test/auth.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+const chai = require('chai')
+const expect = chai.expect
+const mojang = require('../')
+
+describe('mojang.auth()', () => {
+
+  // TODO implement test scenario with valid credentials
```
Comment (jamen) [inline test/auth.js]: For example, before you make the `mojang.auth(...)` calls, setup `nock('https://authserver.mojang.com')` mocks.

**Comment by modev** [comment]
@jamen  I implemented the HTTP interceptor only for the auth endpoint. I think actually requesting the other resources from the Mojang API helps to recognize breaking changes faster.

**Comment by jamen** [comment]
:+1:

You don't need to mock all the requests,  just where it makes sense.

**Comment by modev** [comment]
Tests are done so far. If I'm missing anything please comment. 
But I'm getting a really annoying bug with the _user_ test. It seems to be crashing the username test. I tried to fix this by allowing unmocked requests to bypass the interceptor, but it's still not working. @jamen Any ideas how to fix this?

File: test/history.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        expect(history).to.have.length.of.at.least(1)
+        done()
+      })
+      .catch((err) => {
+        expect(err).to.be.null
+        done()
+      })
+  })
+
+  it('with an invalid uuid should reject with error', () => {
```
Comment (jamen) [inline test/history.js]: You forgot `(done) => ...` here, getting an undefined error.

**Review by jamen — CHANGES_REQUESTED** [review]

File: test/user.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+'use strict';
+const chai = require('chai')
+const expect = chai.expect
+const nock = require('nock')
+const mojang = require('../')
+
+describe('mojang.user()', () => {
```
Comment (jamen) [inline test/user.js:7]: Maybe you should setup [`after` or `afterEach`](http://stackoverflow.com/questions/16747367/mocha-js-to-run-after-hook-even-if-test-suit-fails) hooks here (and/or inside `username` tests), that does [`nock.cleanAll()`](https://github.com/node-nock/nock#cleanall)

**Review by jamen — CHANGES_REQUESTED** [review]

**Comment by jamen** [comment]
Looks good nice work!  I enabled Travis for you.

**Review by jamen — APPROVED** [review]

**Comment by jamen** [comment]
:tada:

**Comment by modev** [comment]
The test hooks fixed the error. Travis builds the project without any errors. 👍 🎉

**Pull request merged**